lib/signaling/v2/iceconnectionmonitor.js

```diff
@@ -0,0 +1,128 @@
+'use strict';
+
+const { ICE_ACTIVITY_CHECK_PERIOD_MS, ICE_INACTIVITY_THRESHOLD_MS } = require('../../util/constants');
+
+/**
+ * Monitors a {@link RTCPeerConnection}'s media stats and notifies
+ * caller when inactivity is detected.
+ */
+class IceConnectionMonitor {
+  /**
+   * Construct an {@link IceConnectionMonitor}.
+   * @param {RTCPeerConnection} peerConnection
+   * @param {object} [options]
+   */
+  constructor(peerConnection, options) {
+    options = Object.assign({
+      activityCheckPeriodMs: ICE_ACTIVITY_CHECK_PERIOD_MS,
+      inactivityThresholdMs: ICE_INACTIVITY_THRESHOLD_MS,
+    }, options);
+
+    Object.defineProperties(this, {
+      _activityCheckPeriodMs: {
+        value: options.activityCheckPeriodMs
+      },
+      _inactivityThresholdMs: {
+        value: options.inactivityThresholdMs
+      },
+      _lastActiveStats: {
+        value: null,
+        writable: true
+      },
+      _peerConnection: {
+        value: peerConnection
+      },
+      _timer: {
+        value: null,
+        writable: true,
+      }
+    });
+  }
+
+  // TODO: remove this before merging.
+  _makarandlog() {
+    var newArgs = ['makarand: '];
+    var args = Array.prototype.slice.call(arguments);
+    console.log.apply(console.log, newArgs.concat(args));
+  }
+
+  /**
+   * Get ICE connection stats.
+   * @returns Promise<{?RTCIceCandidatePairStats}>
+   */
+  _getIceConnectionStats() {
+    return this._peerConnection.getStats().then(stats => Array.from(stats.values()).find(stat => {
+      return stat.type === 'candidate-pair' && stat.nominated;
+    }));
+  }
+
+  /**
+   * returns ICE connection stats or null.
+   * @returns Promise<?{timestamp, bytesReceived, bytesSent}>
+   */
+  _getStats() {
```

Comment:
@makarandp0 I still think we don't need this method. Why would we want to copy the stats values into a JSON instead of just using the RTCIceCandidateStats directly? You can also attach the `catch()` handler to `_getIceConnectionStats()`.

Reply:
okay, I guess the method has served its purpose of being able to test. I will removed it and fold into `_getIceConnectionStats`